src/main/java/org/jabref/logic/util/io/BackupFileUtil.java

```diff
@@ -48,12 +48,13 @@ public static Path getAppDataBackupDir() {
      *     (and configured in the preferences as "make backups")
      * </p>
      */
-    public static Path getPathForNewBackupFileAndCreateDirectory(Path targetFile, BackupFileType fileType) {
+
+    public static Path getPathForNewBackupFileAndCreateDirectory(Path targetFile, BackupFileType fileType, Path backupDir) {
         String extension = "." + fileType.getExtensions().get(0);
         String timeSuffix = ZonedDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd--HH.mm.ss"));
 
         // We choose the data directory, because a ".bak" file should survive cache cleanups
-        Path directory = getAppDataBackupDir();
+        Path directory = backupDir;
```

Comment:
Can you rename the local variable `directory` to `backupDir` and remove this statement?